### PR TITLE
158 Rebased refactor compute lumped mass

### LINKED ISSUE #158: Refactor Computation of Lumped Mass
To get one executable, we refactor the computation of the lumped mass.

### PR DESCRIPTION
**Pull request by uhetmaniuk**
Closes #158

### CODE REVIEW

File: src/nimble_data_manager.cc [hunk 1]
```diff
@@ -65,40 +66,74 @@ DataManager::DataManager(const nimble::Parser &parser,
...
+
   field_ids_.reference_coordinates = macroscale_data_->AllocateNodeData(nimble::VECTOR, "reference_coordinate", num_nodes);
   field_ids_.displacement = macroscale_data_->AllocateNodeData(nimble::VECTOR, "displacement", num_nodes);
   field_ids_.velocity = macroscale_data_->AllocateNodeData(nimble::VECTOR, "velocity", num_nodes);
   field_ids_.acceleration = macroscale_data_->AllocateNodeData(nimble::VECTOR, "acceleration", num_nodes);
   field_ids_.internal_force = macroscale_data_->AllocateNodeData(nimble::VECTOR, "internal_force", num_nodes);
+
   field_ids_.contact_force = macroscale_data_->AllocateNodeData(nimble::VECTOR, "contact_force", num_nodes);
 
   if (!parser_.UseKokkos()) {
```
Comment (uhetmaniuk) [inline src/nimble_data_manager.cc:128]: The next variables are only used in the quasi-static case.

File: src/nimble_kokkos.cc [hunk 2]
```diff
@@ -705,7 +663,7 @@ int ExplicitTimeIntegrator(nimble::Parser & parser,
 
     // Perform a reduction to obtain correct values on MPI boundaries
     watch_internal.push_region("MPI reduction");
-    myVectorCommunicator.VectorReduction(mpi_vector_dimension, internal_force_h);
+    myVectorCommunicator->VectorReduction(mpi_vector_dim, internal_force_h);
```
Comment (uhetmaniuk) [inline src/nimble_kokkos.cc:666]: Use a constant variable for 'mpi_vector_dim'. 
Ideally we should use the spatial dimension but, currently, 'internal_force_h' (and others) are double*[3] (or a Kokkos equivalent).

File: src/nimble_data_manager.cc (hunk 1 above)
Comment (uhetmaniuk) [reply src/nimble_data_manager.cc:128]: @rjones-sandia Comment for https://github.com/NimbleSM/NimbleSM/pull/159#pullrequestreview-605323423

File: src/nimble_kokkos.cc (hunk 2 above)
Comment (uhetmaniuk) [reply src/nimble_kokkos.cc:666]: @rjones-sandia Comment to address https://github.com/NimbleSM/NimbleSM/pull/159#pullrequestreview-605323423

**Review by rjones-sandia — APPROVED** [review]
thanks for being careful and patient with the refactor

I think I see that ModelData will be the/one of the vehicles for collapsing the various flavors of main.cc, so it is transitioning from an allocator/data manager to also compute quantities.

Basic info questions 
Will more "compute" functionality be added to ModelData?
Will there be a single "main" or will we still need multiple (but slimmer) mains for the various flavors of parallel?

**Pull request merged**